install/Knative-with-Minishift.md

```diff
@@ -0,0 +1,250 @@
+```
+
+1. Run the following to install Istio:
+
+    ```shell
+    curl -L https://storage.googleapis.com/knative-releases/serving/latest/istio.yaml \
+    | oc apply -f -
+    ```
+    > **NOTE:** If you get a lot of errors after running the above command like:  __unable to recognize "STDIN": no matches for kind "Gateway" in version "networking.istio.io/v1alpha3"__, just run the command above again, it's idempotent and hence objects will be created only once.
+
+2. Ensure the istio-sidecar-injector pods runs as provileged:
+    ```shell
+    oc get cm istio-sidecar-injector -n istio-system -o yaml  \
+     | sed -e 's/securityContext:/securityContext:\\n      privileged: true/' \
+     | oc replace -f -
+    ```
+3. Monitor the Istio components until all of the components show a `STATUS` of `Running` or `Completed`:
+
+    ```shell
+    while oc get pods -n istio-system | grep -v -E "(Running|Completed|STATUS)"; do sleep 5; done
+    ```
+> **NOTE:** It will take a few minutes for all the components to be up and running. Use CTRL+C to exit watch mode
+
+## Install Knative Serving
+
+The following section details on deploying [Knative Serving](https://github.com/knative/serving) to OpenShift.
+
+The [knative-openshift-policies.sh](scripts/knative-openshift-policies.sh) runs the required commands to configure necessary privileges to the service accounts used by Knative.
+
+```shell
+bash <(curl -s https://raw.githubusercontent.com/knative/docs/master/install/scripts/knative-openshift-policies.sh)
+```
+
+1. Install Knative serving:
+
+    ```shell
+     oc apply -f https://storage.googleapis.com/knative-releases/serving/latest/release-no-mon.yaml
+    ```
+
+2. Monitor the Knative components until all of the components show a `STATUS` of `Running` or `Completed`:
+
+    ```shell
+    while oc get pods -n knative-build | grep -v -E "(Running|Completed|STATUS)"; do sleep 5; done
+    while oc get pods -n knative-serving | grep -v -E "(Running|Completed|STATUS)"; do sleep 5; done
+    ```
+    The first command watches for all pod status in `knative-build` and the second command will watch for all pod status in `knative-serving`.
+
+3. Set route to access the  OpenShift ingress CIDR, so that services can be accessed via LoadBalancerIP
+    ```shell
+    # Only for macOS
+    sudo route -n add -net $(minishift openshift config view | grep ingressIPNetworkCIDR | awk '{print $NF}') $(minishift ip)
+    # Only for Linux
+    sudo ip route add $(minishift openshift config view | grep ingressIPNetworkCIDR | sed 's/\r$//' | awk '{print $NF}') via $(minishift ip)
+    ```
+> **NOTE:** It will take a few minutes for all the components to be up and running. Use CTRL+C to exit watch mode.
+
+## Deploying an app
+
+Now that your cluster has Knative installed, you're ready to deploy an app.
+
+1. Deploy sample HelloWorld app:
+
+    ```shell
+    oc project myproject
+    echo '
+    apiVersion: serving.knative.dev/v1alpha1 # Current version of Knative
```

Comment:
```
apiVersion: serving.knative.dev/v1alpha1 # Current version of Knative
kind: Service
metadata:
  name: helloworld-go # The name of the app
spec:
  runLatest:
    configuration:
      revisionTemplate:
        spec:
          container:
            image: gcr.io/knative-samples/helloworld-go # The URL to the image of the app
            env:
            - name: TARGET # The environment variable printed out by the sample app
              value: "Go Sample v1"

```

instead

Reply:
@kameshsampath that's what worked, the original _inlined_ yaml not - do to space issues

Reply:
damn!!! i was always afraid of it